security.md

```diff
@@ -0,0 +1,94 @@
+due to their disrupting nature.
+
+In short, guessing codes is unlikely to be an attractive target for any attacker.
+Of course, there is always the possibility to configure a longer code in the
+clients if desired.
+
+## Denial of Service (DoS) attacks on the Rendezvous server
+
+Wormhole codes can be so short because they implicitly contain a common
+rendezvous server URL (any two applications that use Magic Wormhole
+should be configured to use the same server). As a result, successful
+operation depends upon both clients being able to contact that server,
+making it a single point of failure (SPOF).
+
+While it is of course possible to incapacitate the rendezvous server using
+traditional means of DoS, so much work is not necessary: An attacker may
+simply connect to every nameplate with a random code, causing the key exchange
+to fail. There is a "list" command in the protocol which makes it easy to
+enumerate all nameplates to disrupt them, but even without it the name space is
+sufficiently small (by design, as we want short codes) to brute force.
+
+There are several ideas to defend against this, however the only one already
+specified in the protocol is the "permission" feature: The rendezvous server
+may use it to restrict new connections with some kind of challenge (or for
+private servers, authentication). The motivation behind this is to enable
+a proof-of-work challenge (using HashCash) when under attack to slow it down.
+See the server protocol specification for details.
+
+Currently the best defense against this is to use high-entropy nameplates which
+cannot easily be enumerated at the cost of longer codes. This has not be
+implemented yet though, and would still leave the attack vector of "ordinary"
+DoS.
+
+## Metadata leaks
+
+As usual in cryptography, communication metadata cannot easily be protected. Any
+involved server will know the timestamp and size of each message it relays. This
+usually is not much of an issue, except for file transfer where the file size may
+be characteristic for its content. (Imagine people using Magic Wormhole to send
+some specific movie.)
+
+Thanks to the separation between rendezvous server and relay server, forcing the
+client to establish a direct connection will bypass the relay server and thus not
+leak such metadata for file transfers.
+
+## IP address leakage and anonymity
+
+As usual on the internet, any involved server will know connection information
+including the IP address of any two peers.
+
+Additionally, in order to establish a direct connection, public and local IP
+addresses are sent to the peer. This may leak information about the local
+network topology, including potentially used VPNs. If you do not trust your
+peer with such information, enforcing a relay server connection is advised.
+However, keep in mind that it is easy to host a relay server, so it might still
+leak your public IP address.
+
+If you do not trust the servers with knowing your IP address either, use
```

Comment:
We generally try to avoid this word .. better is "expose"; that is something like "if you do not want to expose your IP to ..."

Reply:
Which word?

Reply:
Ah, sorry ;) "trust"

Reply:
Oh that's interesting, why? Because I see trust as an essential part of threat modeling and assessing individual security risks.

Reply:
I think this was via Tor people I'm thinking of? You aren't actually "trusting" the server, so the "expose" I guess is trying to get at what's actually happening. That is, that you reveal your IP address to them -- I don't think "trust" enters into it per se. Maybe "reveal" is a good one too?

I believe the logic is that it doesn't matter if you "trust" the server, or the people running the server -- the important thing is what information you reveal. For example, even if they are trustworthy they can themselves be compromised or forced to reveal any information they have (subpoena, etc)